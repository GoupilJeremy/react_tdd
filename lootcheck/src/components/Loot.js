import React, { Component } from 'react';
import { connect } from 'react-redux';
import { fetchBitcoin } from '../actions/bitcoin';

export class Loot extends Component {
    componentDidMount() {
        this.props.fetchBitcoin();
    }
    render() {
        return (
            <h3>Bitcoin balance:</h3>
        )
    }
}

export default connect(state => state, { fetchBitcoin })(Loot);