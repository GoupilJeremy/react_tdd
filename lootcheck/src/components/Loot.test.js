import React from 'react';
import { mount, shallow } from 'enzyme';
import { Loot } from './Loot';

describe('Loot', () => {
    const mockFetchBitcoin = jest.fn();
    let props = { balance: 10, bitcoin: {}, fetchBitcoin: mockFetchBitcoin};
    let loot = shallow(<Loot {...props}/>);

    it('renders properly', () => {
        expect(loot).toMatchSnapshot();
    });

    describe('when mounted', () => {
        beforeEach(() => {
            props.fetchBitcoin = mockFetchBitcoin;
            loot = mount(<Loot {...props} />);
        });

        it('dispatches the `fetchBitcoin` method it receives from props', () => {
            expect(mockFetchBitcoin).toHaveBeenCalled();
        });
    });

    describe('when there are valid bitcoin props', () => {
        beforeEach(() => {
            props = { balance: 10, bitcoin: { bpi: {USD: { rate: '1,000'}}}};
            loot = shallow(<Loot {...props}/>);
        });

        it('displays the correct bitcoin value', () => {
            expect(loot.find('h3').text()).toEqual('Bitcoin balance: 0.01');
        });
    });
});
