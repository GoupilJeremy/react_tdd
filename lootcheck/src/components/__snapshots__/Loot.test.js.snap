// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loot renders properly 1`] = `ShallowWrapper {}`;
